src/processors/uploadAvatarToIpfs/processor.ts

```diff
@@ -0,0 +1,130 @@
+  clients: Clients,
+): Promise<UploadedAvatarDetails> => {
+  console.log('uploadAvatarToIpfs:: uploading ', profileWithFile.avatarUrl);
+
+  if (profileWithFile.avatarUrl == null) {
+    throw new Error(`Error uploading avatar: the URL to upload is null!`);
+  }
+
+  // Check if this file is already on IPFS -- if so, we don't need to upload it ourselves,
+  // just to record its hash in the DB.
+  const preexistingIpfsHash = extractHashFromURL(profileWithFile.avatarUrl);
+  if (preexistingIpfsHash) {
+    // Return everything that is needed to update both DB entries: artist profile, and IPFS file
+    return {
+      ...extractArtistProfileId(profileWithFile),
+      avatarUrl: profileWithFile.avatarUrl,
+      cid: preexistingIpfsHash,
+    };
+  }
+
+  try {
+    const file = await clients.ipfs.client.add(
+      urlSource(profileWithFile.avatarUrl),
+      {
+        pin: false,
+        timeout: parseInt(process.env.IPFS_API_TIMEOUT!),
+      },
+    );
+
+    // Return everything that is needed to update both records: artist profile, and IPFS file
+    return {
+      ...extractArtistProfileId(profileWithFile),
+      avatarUrl: profileWithFile.avatarUrl,
+      cid: file.cid.toString(),
+    };
+  } catch (e: any) {
+    throw new IPFSUploadError(e.message, profileWithFile.avatarUrl);
+  }
+};
+
+/**
+ * Adds avatar images for artist profiles to IPFS. Saves the CID of these newly-uploaded avatar images to the database as avatarIPFSHash.
+ * @param artistProfilesWithFiles
+ * @param clients
+ */
+const processorFunction = async (
+  artistProfilesWithFiles: ProfileWithAvatarIpfsFile[],
+  clients: Clients,
+) => {
+  const results = await rollPromises<
+    ProfileWithAvatarIpfsFile,
+    UploadedAvatarDetails,
+    IPFSUploadError
+  >(
+    artistProfilesWithFiles,
+    profileIpfsFile => uploadAvatar(profileIpfsFile, clients),
+    300,
+    10000,
+  );
+
+  const uploadedAvatarDetails = results
+    .filter(result => !result.isError)
+    .map(result => result.response) as UploadedAvatarDetails[];
+  // We need to assert this type above because TypeScript isn't smart enough to know that this
+  // filters out all error responses
+
+  const artistProfileUpdates = uploadedAvatarDetails.map(
+    ({artistId, platformId, cid}) => ({
+      artistId,
+      platformId,
+      avatarIPFSHash: cid,
+    }),
+  );
+
+  const ipfsFileSuccessUpdates = uploadedAvatarDetails.map(
+    ({avatarUrl, cid}) => ({url: avatarUrl, cid}),
+  );
+
+  const ipfsFileFailureUpdates = results
+    .filter(result => result.isError)
+    .map(result => result.error as IPFSUploadError)
+    .map(error => ({url: error.url, error: error.message}));
+
+  const ipfsFileUpdates = [
+    ...ipfsFileSuccessUpdates,
+    ...ipfsFileFailureUpdates,
+  ];
+
+  if (ipfsFileFailureUpdates.length > 0) console.log("IPFS file uploads failed: ", JSON.stringify(ipfsFileFailureUpdates))
+
+  // TODO: update rather than upsert (needs change to db client to accept multiple IDs)
```

Comment:
yeah. i think can just remove this comment for now since we're tracking this upsert issue elsewhere already